privacyidea/lib/token.py

```diff
@@ -1128,16 +1127,13 @@ def set_defaults(serial):
     :type serial: basestring
     :return: None
     """
-    tokenobject_list = get_tokens(serial=serial)
-    if tokenobject_list:
-        db_token = tokenobject_list[0].token
-        db_token.otplen = int(get_from_config("DefaultOtpLen", 6))
-        db_token.count_window = int(get_from_config("DefaultCountWindow", 15))
-        db_token.maxfail = int(get_from_config("DefaultMaxFailCount", 15))
-        db_token.sync_window = int(get_from_config("DefaultSyncWindow", 1000))
-        db_token.tokentype = u"hotp"
-        db_token.save()
-
+    db_token = get_one_token(serial=serial).token
+    db_token.otplen = int(get_from_config("DefaultOtpLen", 6))
+    db_token.count_window = int(get_from_config("DefaultCountWindow", 15))
+    db_token.maxfail = int(get_from_config("DefaultMaxFailCount", 15))
+    db_token.sync_window = int(get_from_config("DefaultSyncWindow", 1000))
+    db_token.tokentype = u"hotp"
+    db_token.save()
```

Comment:
The logic now is very API centric.
The library call "set_defaults" will now raise an exception is the serial does not exist. This is fine for an API call but may result in unexpected behaviour, if ``set_detaults`` is called from without other code.
(justsaying)

Reply:
Works for me.